docx.go

```diff
@@ -52,32 +68,59 @@ func ConvertDocx(r io.Reader) (string, map[string]string, error) {
 					meta["CreatedDate"] = fmt.Sprintf("%d", t.Unix())
 				}
 			}
-
-		case f.Name == "word/document.xml":
-			textBody, err = parseDocxText(f)
+		case override.ContentType == "application/vnd.openxmlformats-officedocument.wordprocessingml.document.main+xml":
+			body, err := parseDocxText(f)
 			if err != nil {
 				return "", nil, err
 			}
-
-		case reHeaderFile.MatchString(f.Name):
-			header, err := parseDocxText(f)
+			textBody += body + "\n"
+		case override.ContentType == "application/vnd.openxmlformats-officedocument.wordprocessingml.footer+xml":
+			footer, err := parseDocxText(f)
 			if err != nil {
 				return "", nil, err
 			}
-			textHeader += header + "\n"
-
-		case reFooterFile.MatchString(f.Name):
-			footer, err := parseDocxText(f)
+			textFooter += footer + "\n"
+		case override.ContentType == "application/vnd.openxmlformats-officedocument.wordprocessingml.header+xml":
+			header, err := parseDocxText(f)
 			if err != nil {
 				return "", nil, err
 			}
-			textFooter += footer + "\n"
+			textHeader += header + "\n"
 		}
-	}
 
+	}
 	return textHeader + "\n" + textBody + "\n" + textFooter, meta, nil
 }
 
+func getContentTypes(f *zip.File) (*Type, error) {
+	contentTypesFile, err := f.Open()
+	if err != nil {
+		return nil, err
+	}
+	defer contentTypesFile.Close()
+
+	contentTypesFileBytes, err := ioutil.ReadAll(contentTypesFile)
+	if err != nil {
+		return nil, err
+	}
+
+	var types Type
+	err = xml.Unmarshal(contentTypesFileBytes, &types)
+	if err != nil {
+		return nil, err
+	}
+	return &types, nil
+}
+
+func mapZipFiles(files []*zip.File) map[string]*zip.File {
+	filesMap := map[string]*zip.File{}
```

Comment:
I would put the size in here too (`2 * len(files)`)

Reply:
done.